src/region.rs

```diff
@@ -65,29 +65,29 @@ impl TryFrom<RegionOrEndpoint> for Region {
 /// Translate an endpoint URL into a Region
 pub fn region_from_endpoint(endpoint: &str) -> Result<Region, ParseError> {
     let uri = endpoint.parse::<Uri>().context(EndpointParseError)?;
-    let name = region_name_from_host(uri.host().unwrap_or(""));
+    let name = region_name_from_host(uri.host().unwrap_or(""))
+        .unwrap_or_else(|| Region::default().name().into());
```

Comment:
**nit**: Could this be something like
```rust
let name = uri.host()
    .map(region_name_from_host)
    .unwrap_or_else(|| Region::default().name().into());
```
The flow is a little bit funky right now with the two unwraps and passing `""` to `region_name_from_host`.

Reply:
Oh, yeah, definitely, although it needs to be `and_then` instead of `map`. The logic above grew out of a simpler expression in a couple of steps.